File: 1711-FindValidMatrixGivenRowAndColumnSums/1711-FindValidMatrixGivenRowAndColumnSums.java
// Last updated: 3/26/2026, 1:48:05 AM
class Solution {
    public int[][] restoreMatrix(int[] rowSum, int[] colSum) {
        int[][] matrix = new int[rowSum.length][colSum.length];
        int i=0;
        int j=0;
        int row = rowSum.length;
        int col = colSum.length;
        while(i<row && j<col){
            int min = Math.min(rowSum[i],colSum[j]);
            matrix[i][j] = min;
            rowSum[i] = Math.abs(rowSum[i]-min);
            colSum[j] = Math.abs(colSum[j]-min);
            if(rowSum[i]==0)
                i++;
            else
                j++;
        }
        return matrix;
    }
}

File: 1711-FindValidMatrixGivenRowAndColumnSums/1711-FindValidMatrixGivenRowAndColumnSums.cs
// Last updated: 3/26/2026, 1:48:02 AM
public class Solution {
    public int[][] RestoreMatrix(int[] rowSum, int[] colSum) {
        int m = rowSum.Length, n = colSum.Length;
            int[][] grid = new int[m][];
            int i =0,j=0;
            for(int k = 0;k<grid.Length;k++)
            {
                grid[k] = new int[n];
            }
            while( i < m && j < n ){
                if(rowSum[i]<colSum[j])
                {
                    grid[i][j] = rowSum[i];
                    colSum[j] -= rowSum[i];
                    i++;
                }
                else if(rowSum[i]>=colSum[j])
                {
                    grid[i][j] = colSum[j];
                    rowSum[i] -= colSum[j];
                    j++;
                }
            }
            return grid;
    }
}